pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -104,20 +101,41 @@ private SchemaType getSchemaTypeOrDefault(String topic, Class<?> clazz) {
     }
 
     private static SchemaType getDefaultSchemaType(Class<?> clazz) {
-        if (byte[].class.equals(clazz)
-            || ByteBuf.class.equals(clazz)
-            || ByteBuffer.class.equals(clazz)) {
-            return SchemaType.NONE;
+        if (byte[].class.equals(clazz)) {
+            return BytesSchema.of().getSchemaInfo().getType();
```

Comment:
why don't we return `SchemaType.BYTES` like what other branches are doing? at least, we need to make them consistent.